### PR TITLE
Added redirect by page name feature to the redirect tag and a test for it.

### LINKED ISSUE #2546: Add redirect tag feature by page name
Hi, this is a [first-timers-only issue](https://publiclab.github.io/community-toolbox/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

We have a redirect system that uses tags -- read more at https://publiclab.org/wiki/power-tags -- but it requires tags to point at a "node id" like `redirect:12345`. This is hard to remember and kind of obscure -- so let's make it possible to use a page "slug" as well. Then you could redirect to the page at `/wiki/openhour` by using `redirect:openhour`. 

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [app/controllers/wiki_controller.rb](https://github.com/publiclab/plots2/blob/master/wiki_controller.rb) in the `plots2` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://publiclab.github.io/community-toolbox/#r=all) for some help in taking your first steps!

Below is a "diff" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add:

```diff
@@ -33,6 +33,14 @@ def show
         flash.now[:warning] = "Only moderators and admins see this page, as it is redirected to <a href='#{Node.find(@node.power_tag('redirect')).path}'>#{Node.find(@node.power_tag('redirect')).title}</a>.
         To remove the redirect, delete the tag beginning with 'redirect:'"
       end
+    elsif @node&.has_power_tag('redirect') && Node.where(slug: @node.power_tag('redirect')).exists?
+      if current_user.nil? || !current_user.can_moderate?
+        redirect_to Node.find_by(slug: @node.power_tag('redirect')).path
+        return
+      elsif current_user.can_moderate?
+        flash.now[:warning] = "Only moderators and admins see this page, as it is redirected to <a href='#{Node.find_by(slug: @node.power_tag('redirect')).path}'>#{Node.find_by(slug: @node.power_tag('redirect')).title}</a>.
+        To remove the redirect, delete the tag beginning with 'redirect:'"
+      end
     end
 
     if @node&.has_power_tag('abtest') && !Node.where(nid: @node.power_tag('abtest')).empty?
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### 🤔❓ Questions?

Leave a comment below!

### Is someone else already working on this?

We encourage you to link to this issue by mentioning the issue # in your pull request, so we can see if someone's already started on it. **If someone seem stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://publiclab.github.io/community-toolbox/#r=all). Thanks!

(This issue was created by [First-Timers-Bot](https://github.com/hoodiehq/first-timers-bot).)

### PR DESCRIPTION
**Pull request by r0vina**
Fixes #2546 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

File: app/controllers/wiki_controller.rb [hunk 1]
```diff
@@ -34,7 +34,14 @@ def show
         To remove the redirect, delete the tag beginning with 'redirect:'"
       end
     end
```
Comment (sagarpreet-chadha) [inline app/controllers/wiki_controller.rb]: Hi @r0vina !   
I guess there is this extra `end` here . Apart from this all looks perfect !!!
Great 😄 !

**Comment by sagarpreet-chadha** [comment]
This looks perfect to me :-)
@r0vina ...you can find more issues [here](https://publiclab.github.io/community-toolbox/#r=all)  if you are interested , we would love to have your help on them. Thanks !
And Welcome to PublicLab  😄 !

**Comment by jywarren** [comment]
This looks perfect! 

Do you think you'd be interested in adding a functional test for this expansion of the redirect feature? It could mirror this one, but using, for example, `redirect:about`:

https://github.com/publiclab/plots2/blob/c038e04449d3f9f99ecad2b42352aec790d38331/test/functional/wiki_controller_test.rb#L522-L531

**Comment by r0vina** [comment]
I could certainly try, although I only have second-hand experience with C++ unit tests.

**Comment by jywarren** [comment]
This is fantastic! Thanks so much, it's a great test. We'd love to have more help from you if you're interested!

**Pull request merged**